src/dmi.rs

```diff
@@ -63,3 +68,32 @@ fn create_png(path: &str, width: &str, height: &str, data: &str) -> Result<()> {
     let mut writer = encoder.write_header()?;
     Ok(writer.write_image_data(&result)?)
 }
+
+#[derive(Debug)]
+enum ResizePngError {
+    StringParse(std::num::ParseIntError),
+    Image(ImageError),
+}
+
+impl From<std::num::ParseIntError> for ResizePngError {
+    fn from(error: std::num::ParseIntError) -> Self {
+        ResizePngError::StringParse(error)
+    }
+}
+
+impl From<ImageError> for ResizePngError {
+    fn from(error: ImageError) -> Self {
+        ResizePngError::Image(error)
+    }
+}
+
+fn resize_png(path: &str, width: &str, height: &str) -> std::result::Result<(), ResizePngError> {
+    let width = u32::from_str_radix(width, 10)?;
+    let height = u32::from_str_radix(height, 10)?;
+
+    let img = image::open(path)?;
+
+    let newimg = img.resize(width, height, image::imageops::Nearest);
```

Comment:
Wouldn't it be better to allow choice of resizing method? Nearest Neighbor isn't always the best.